packages/forma-36-react-components/src/components/Select/Select.tsx

```diff
@@ -38,95 +36,84 @@ const defaultProps: Partial<SelectProps> = {
   willBlurOnEsc: true,
 };
 
-export class Select extends Component<SelectProps, SelectState> {
-  static defaultProps = defaultProps;
+export const Select = (props: SelectProps) => {
+  const {
+    id,
+    name,
+    required,
+    children,
+    width,
+    className,
+    testId,
+    onChange,
+    onBlur,
+    onFocus,
+    isDisabled,
+    hasError,
+    value,
+    willBlurOnEsc,
+    ...otherProps
+  } = props;
+  const [valueState, setValueState] = useState<string | undefined>();
 
-  state = {
-    value: this.props.value,
-  };
+  const handleKeyDown = (e: KeyboardEvent<HTMLSelectElement>) => {
+    const ESC = 27;
 
-  UNSAFE_componentWillReceiveProps(nextProps: SelectProps) {
-    if (this.props.value !== nextProps.value) {
-      this.setState({
-        value: nextProps.value,
-      });
+    if (e.keyCode === ESC && willBlurOnEsc) {
+      e.currentTarget.blur();
     }
-  }
+  };
 
-  handleKeyDown = (e: KeyboardEvent<HTMLSelectElement>) => {
-    const ESC = 27;
+  const handleChange = (e: ChangeEvent<HTMLSelectElement>) => {
```

Comment:
I think we should wrap all event handlers in `useCallback` just for good measure.